aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -141,7 +141,7 @@ Tensor& empty_out(Tensor& result, IntArrayRef size) {
     return target_type.copy(self, non_blocking);                 \
   }
 
-AT_FORALL_SCALAR_TYPES(DEFINE_CAST_OP)
+AT_FORALL_SCALAR_TYPES_AND_BOOL(DEFINE_CAST_OP)
```

Comment:
I wonder if it's possible to adjust our `AT_FORALL` macros in the same way we adjusted the `AT_DISPATCH` macros. Maybe this is not conveniently possible.

Reply:
I think we can -- we don't need to worry about backwards incompatibility either!

Reply:
will fix as a separate PR